src/manager.ts

```diff
@@ -97,36 +69,65 @@ const getCurrentFirmware = async (
   provider: number,
 ): Promise<FinalFirmware> => {
   const url = new URL(
-    `wss://scriptrunner.api.live.ledger.com/update/get_device_version`,
+    `https://manager.api.live.ledger.com/api/get_firmware_version`,
   );
-
-  const {
-    data,
-  }: {
+  const { data }: {
     data: FinalFirmware;
-  } = await axios.post(url.toString(), {
-    device_version: deviceId,
-    version_name: version,
-    provider: provider,
+  } = await axios(url.toString(), {
+    method: "POST",
+    headers: {
+      'Content-type': 'application/x-www-form-urlencoded',
+    },
+    data: JSON.stringify({
+      device_version: deviceId,
+      version_name: version,
+      provider: 1,
+    })
   });
   return data;
 };
 
+export const installApp = async (
+  app,
+  transport: Transport,
+  isDelete: boolean,
+): Promise<void> => {
```

Comment:
maybe return a value to tell when installApp has finish ?

Reply:
I don't think we can. The WS seems to close before the install even start. it might be listenable on transport instead